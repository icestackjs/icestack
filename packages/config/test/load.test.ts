import path from 'node:path'
import { loadSync } from '@/.'
import ci from 'ci-info'

describe.skipIf(ci.isCI)('load config', () => {
  it('load ts config', () => {
    const config = loadSync({
      cwd: path.resolve(__dirname, 'ts'),
    })
    expect(config?.config.log).toBe(false)
    expect(config.config).toMatchSnapshot()
  })

  it('load js config', () => {
    const config = loadSync({
      cwd: path.resolve(__dirname, 'js'),
    })
    expect(config?.config.log).toBe(false)
    expect(config.config).toMatchSnapshot()
  })
  // For the synchronous API, the only difference is that .mjs files are not included. See "Loading JS modules" for more information.
  // https://www.npmjs.com/package/cosmiconfig?activeTab=readme
  it.skip('load mjs config', () => {
    const config = loadSync({
      cwd: path.resolve(__dirname, 'esm'),
    })
    expect(config?.config.log).toBe(false)
    expect(config.config).toMatchSnapshot()
  })

  it('load cjs config', () => {
    const config = loadSync({
      cwd: path.resolve(__dirname, 'cjs'),
    })
    expect(config?.config.log).toBe(false)
    expect(config.config).toMatchSnapshot()
  })
})
