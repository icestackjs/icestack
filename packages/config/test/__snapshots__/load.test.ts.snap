// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`load config > load cjs config 1`] = `
{
  "config": {
    "base": {
      "themes": {
        "dark": {
          "selector": ".dark",
        },
        "light": {
          "selector": "page",
        },
      },
    },
    "log": false,
  },
  "filepath": "/Users/icebreaker/Documents/GitHub/icestack/packages/config/test/cjs/icestack.config.cjs",
}
`;

exports[`load config > load js config 1`] = `
{
  "config": {
    "base": {
      "themes": {
        "dark": {
          "selector": ".dark",
        },
        "light": {
          "selector": "page",
        },
      },
    },
    "log": false,
  },
  "filepath": "/Users/icebreaker/Documents/GitHub/icestack/packages/config/test/js/icestack.config.cjs",
}
`;

exports[`load config > load ts config 1`] = `
{
  "config": {
    "base": {
      "themes": {
        "dark": {
          "selector": ".dark",
        },
        "light": {
          "selector": "page",
        },
      },
    },
    "log": false,
  },
  "filepath": "/Users/icebreaker/Documents/GitHub/icestack/packages/config/test/ts/icestack.config.ts",
}
`;
